src/components/annotations/draw_arrow_head.js

```diff
@@ -51,6 +58,13 @@ module.exports = function drawArrowHead(el3, ends, options) {
 
         startRot = Math.atan2(dy, dx);
         endRot = startRot + Math.PI;
+        if(backOff && startBackOff) {
+            if(backOff * backOff + startBackOff * startBackOff > dx * dx + dy * dy) {
```

Comment:
I don't think this condition is quite right... should be the square of the sum, not the sum of squares. Probably we should just compare lengths, not lengths squared, that was an optimization on my part but it's obviously not very intuitive or extensible!

But also I think we can simplify and only do this whole check once, instead of once for both, then again for `backOff`, and a third time for `startBackOff`. Looks like no matter what both values will be non-negative numbers, so we should be able to do something like:

```js
if(backOff || startBackOff) {
    if(backOff + startBackOff > Math.sqrt(dx * dx + dy * dy) {
        hideline();
        return;
    }
}
```